Add Quarkus documentation (#6138)
open-telemetry/opentelemetry.io: content/en/docs/zero-code/java/_index.md

    otel: 1.46.0
---

Zero-code instrumentation with Java uses a Java agent JAR, Spring Boot

> Starter or Quarkus. To learn how to manually instrument your service or app code, see

Reviewer: ```suggestion
Starter, or Quarkus. To learn how to manually instrument your service or app code, see
```

> Zero-code instrumentation with Java uses a Java agent JAR, Spring Boot Starter,
> or Quarkus. To learn how to manually instrument your service or app code, see

Reviewer 2: ```suggestion
Common options for zero-code instrumentation with Java are the Java agent JAR, Spring Boot Starter,
or the Quarkus OpenTelemetry Extension. To learn how to manually instrument your service or app code, see
```

Can we write this more open ended (since there are options outside these 3, that we are not (yet) listing here)

Author: I think "common options" imply that there might be others. Sounds good enough for me, for now.

    otel: 1.47.0
---

Common options for zero-code instrumentation with Java are the Java agent JAR,

> Spring Boot Starter, or the Quarkus OpenTelemetry Extension. To learn how to

Reviewer: ```suggestion
Spring Boot Starter, and the Quarkus OpenTelemetry Extension. To learn how to
```
